src/source-control/SourceControl.ts

```diff
@@ -43,7 +48,7 @@ export interface SourceControl {
    * @param targetCommit The commit the tree should be rebased on.
    * @returns Promise that resolves when operation is complete.
    */
-  rebaseCommits(rebaseRootCommit: Commit, targetCommit: Commit): Promise<void>;
+  rebaseCommits(rebaseRootCommit: string, targetCommit: string): Promise<void>;
```

Comment:
Why should this be changed to strings? I thought you can get more info from the commit objects

Reply:
Yeah, it's an either/or tbh

Reply:
Actually I agree with your choice of using `string`. That'll allow you to (at some point in the future) handle command line commands more efficiently, as you'll only need to fetch only the commits that are requested, instead of preloading all the `Commit` objects for someone else to filter.

Could you also update the Stacker code that calls this, as well as the frontend code that calls it?